### PR TITLE
Added nullability fixes and tests for hierarchyid abstractions

### PR DESCRIPTION
**Pull request by cutig3r**
Fixes for #11

### CODE REVIEW

**Comment by cutig3r** [comment]
Let me know if this is still too complex.  I still wanted a generic way to get the failure message

File: EFCore.SqlServer.HierarchyId.Test/NullabilityTests.cs [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+{
+    public class NullabilityTests
+    {
+        [Fact]
+        public void Null_equalTo_null_isTrue()
+        {
+            const Operator @operator = Operator.Equal;
+
+            var vals = ((HierarchyId)null, (HierarchyId)null);
+            Assert.True(@operator.Execute(vals), @operator.GetFailureMessage(vals));
```
Comment (smitpatel) [inline EFCore.SqlServer.HierarchyId.Test/NullabilityTests.cs:15]: You don't need operator (or even failure message part).
Just test `Assert.True(((HierarchyId)null) == ((HierarchyId)null));`

Likewise for other operators.
You can also combine like this 
```C#
var left = (HierarchyId)null;
var right = (HierarchyId)null;

Assert.True(left == right);
Assert.False(left != right);
Assert.False(left > right); //and so on for rest
```

Comment (cutig3r) [reply EFCore.SqlServer.HierarchyId.Test/NullabilityTests.cs:15]: ok i'll update those and get rid of the overhead.

**Comment by smitpatel** [comment]
@cutig3r - Thank you for contribution.

**Pull request merged**